### PR TITLE
OOified trip_manage.php

### PR DESCRIPTION
**Pull request by jimbo8098**
Per https://github.com/GibbonEdu/core/pull/1124 

Aside from the general OOification, I've:

- Queries run as gateways now
- Now performing some previously in-code checks in SQL instead which should provide some minor benefits when it comes to efficiency.

### CODE REVIEW

**Comment by jimbo8098** [comment]
Oh, btw, tried setting the namespace to "Gibbon\Module\TripPlanner\Domain" on the gateway but that didn't work out for me. I don't think it's autoloading

File: Trip Planner/trips_manage.php [hunk 1]
```diff
@@ -18,9 +18,12 @@
 */
 
 include "./modules/Trip Planner/moduleFunctions.php";
+include "./modules/Trip Planner/src/Domain/TripPlanner/TripGateway.php";
 
 use Gibbon\Forms\Form;
-
+use Gibbon\Tables\DataTable;
+use Gibbon\Services\Format;
+use Gibbon\Domain\TripPlanner\TripGateway;
```
Comment (SKuipers) [inline Trip Planner/trips_manage.php:26]:
```suggestion
use Gibbon\Module\TripPlanner\Domain;
```

File: Trip Planner/src/Domain/TripPlanner/TripGateway.php [hunk 2]
```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Gibbon\Domain\TripPlanner;
```
Comment (SKuipers) [inline Trip Planner/src/Domain/TripPlanner/TripGateway.php:3]:
```suggestion
namespace Gibbon\Module\TripPlanner\Domain;
```

File: Trip Planner/trips_manage.php [hunk 3]
```diff
@@ -18,9 +18,12 @@
 */
 
 include "./modules/Trip Planner/moduleFunctions.php";
+include "./modules/Trip Planner/src/Domain/TripPlanner/TripGateway.php";
```
Comment (SKuipers) [inline Trip Planner/trips_manage.php:21]:
```suggestion

```

File: Trip Planner/trips_manage.php [hunk 4]
```diff
@@ -165,114 +166,70 @@
...
+              {
+                $actions
+                  ->addAction('approve',__('Approve/Reject'))
+                  ->setURL('/modules/Trip Planner/trips_requestApprove.php')
+                  ->setIcon('iconTick');
+              }
+          });
+        /*
+        $table->modifyRows(function($row) use ($eutFilter,$connection2) {
+            if ($eutFilter) {
```
Comment (SKuipers) [inline Trip Planner/trips_manage.php:223]: Some leftover code from testing? 🤔

File: Trip Planner/trips_manage.php [hunk 5]
```diff
@@ -165,114 +166,70 @@
...
-                            }
-                            if (($row["status"] == "Requested" && needsApproval($connection2, $row["tripPlannerRequestID"], $_SESSION[$guid]["gibbonPersonID"])) == 0 || ($row["status"] == "Awaiting Final Approval" && isApprover($connection2, $_SESSION[$guid]["gibbonPersonID"], true))) {
-                                print "<a href='" . $_SESSION[$guid]["absoluteURL"] . "/index.php?q=/modules/Trip Planner/trips_requestApprove.php&tripPlannerRequestID=" . $row["tripPlannerRequestID"] . "'><img title='" . __m('Approve/Reject') . "' src='./themes/" . $_SESSION[$guid]["gibbonThemeName"] . "/img/iconTick.png'/></a> ";
-                            }
-                        print "</td>";
-                    print "</tr>";
-                    $rowCount++;
+        
+        $tripGateway = $container->get(TripGateway::class);
+        $criteria = $tripGateway->newQueryCriteria()
```
Comment (SKuipers) [inline Trip Planner/trips_manage.php:171]: Add `true` as the parameter here if you'd like to paginate the results.

File: Trip Planner/trips_manage.php [hunk 6]
```diff
@@ -165,114 +166,70 @@
             $result->execute($data);
         } catch (PDOException $e) {
         }
```
Comment (SKuipers) [inline Trip Planner/trips_manage.php:168]: Hmm, is this leftover from the non-ooified code?

**Review by SKuipers — CHANGES_REQUESTED** [review]
Overall looking good. I've added a couple changes and suggestions. The namespace should work in the format of Gibbon\Module\ModuleName

**Comment by jimbo8098** [comment]
I assume your assumption on the domain namespace rename and removal on the include is that it's autoloaded but I wasn't able to do that using the namespace format you suggested (with some amendments):

```
Uncaught Exception: League\Container\Exception\NotFoundException - Alias (Gibbon\Module\TripPlanner\Domain\TripGateway) is not being managed by the container
Line 266 in vendor/league/container/src/Container.php
Line 93 in vendor/league/container/src/Container.php
Line 172 in /home/jimspeir/Projects/Trip-Planner/Trip Planner/trips_manage.php
Line 85 in src/View/View.php
Line 365 in src/View/Page.php
Line 658 in index.php
```
This occurs with the include removed as suggested and the newly formatted use statement:

```
include "./modules/Trip Planner/moduleFunctions.php";
#include "./modules/Trip Planner/src/Domain/TripPlanner/TripGateway.php";

use Gibbon\Forms\Form;
use Gibbon\Tables\DataTable;
use Gibbon\Services\Format;
use Gibbon\Module\TripPlanner\Domain\TripGateway;
#use Gibbon\Module\TripPlanner\Domain;
```
This looks like it did load but can't be referenced within $container->get.

When I used the `use Gibbon\Module\TripPlanner\Domain` I received an error saying that TripGateway wasn't managed by the container but I feel that makes sense considering that's just the namespace.

Cleaned up a lot of the code in this PR a lot in the last commit too with breadcrumbs amended to work using the new style, SQL for SchoolYears being better too although I seem to remember using addSelectSchoolYears in past form ooifications. I see those are part of the DatabaseFormFactory class according to the gibbon docs site but it's not accessible on a normal Form. Could be worthwhile looking into that in another commit.

**Comment by raynichc** [comment]
I thought I'd bump this, as I'm in the process of working on v1.3.00 of the Trip Planner. I'd be happy to merge this into the new branch and make the fixes as necessary if you don't have the time @jimbo8098.

**Comment by jimbo8098** [comment]
I'm not sure given the time that's passed but I believe this is ready for merging

**Comment by raynichc** [comment]
Thanks Jimbo, I've merged it in.

**Comment by SKuipers** [comment]
Thanks Ray and Jim! My apologies, these PRs seem fall off my radar much easier because they're not part of the core. Thanks for bumping the issue & handling the merge. In glancing back at the code, I think it is functional as-is with the file includes. 

It looks like the issue stemed from the file location not matching the file namespace. In this case, the `Trip Planner/src/Domain/TripPlanner/TripGateway.php` has been nested in an extra `TripPlanner` folder, which doesn't match the `Gibbon\Module\TripPlanner\Domain` namespace (which expects TripGateway.php inside the `Trip Planner/src/Domain` folder).

**Pull request merged**